src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -242,7 +242,45 @@ internal int SerializeLogRecord(LogRecord logRecord)
         var timestamp = logRecord.Timestamp;
         var cursor = 0;
         cursor = MessagePackSerializer.WriteArrayHeader(buffer, cursor, 3);
-        cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+
+        var categoryName = logRecord.CategoryName;
+        string eventName = null;
+
+        Span<byte> data = stackalloc byte[0];
+        int validNameLength = 0;
+
+        // If user configured explicit TableName, use it.
+        if (this.m_tableMappings != null && this.m_tableMappings.TryGetValue(categoryName, out eventName))
+        {
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (!this.shouldPassThruTableMappings)
+        {
+            eventName = this.m_defaultEventName;
+            cursor = MessagePackSerializer.SerializeAsciiString(buffer, cursor, eventName);
+        }
+        else if (categoryName.Length > 0)
+        {
+            int cursorStartIdx = cursor;
+            cursor = SanitizeCategoryName(buffer, cursor, ref cursorStartIdx, ref validNameLength, categoryName);
+            if (validNameLength > 0)
+            {
+                data = stackalloc byte[validNameLength + 2];
+                for (int i = 0; i < validNameLength + 2; i++)
```

Comment:
consider using https://docs.microsoft.com/en-us/dotnet/api/system.buffer.blockcopy?view=net-6.0

Reply:
Do we have to copy data from the buffer?

Why not use `buffer.AsSpan().Slice()`?

Reply:
The implementation of Slice() will allocate on the heap which would cause some work from GC in the hotpath.
I am trying to achieve 0 alloc on the heap here with stackalloc.
Related comments:
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/288/files/764bcf516cc471314f2319b496aa3f7b8211a6ce#r850637065
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/288#discussion_r862512696

```
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public Span<T> Slice(int start)
{
    if ((uint)start > (uint)_length)
    {
        ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
    }

    IntPtr byteOffset = _byteOffset.Add<T>(start);
    int length = _length - start;
    return new Span<T>(_pinnable, byteOffset, length);
}
```

Reply:
`Span<T>.Slice` does not cause allocation. The benefit of using Span is that it allows us to avoid allocation and copying.

https://docs.microsoft.com/en-us/dotnet/api/system.span-1?view=net-6.0#spant-and-slices

Reply:
Updated; thank you!

Reply:
You probably forgot to remove the code for copying the bytes.